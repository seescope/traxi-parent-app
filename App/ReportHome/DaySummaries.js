import React, {PropTypes} from 'react';
import {ScrollView, Dimensions} from 'react-native';
import DaySummary from './DaySummary';

const dateComparator = (date1, date2) => date1 > date2 ? -1 : 1;

const sortReports = reports => Object.keys(reports).sort(dateComparator).map(d => [d, reports[d]]);

const {height, width} = Dimensions.get('window');

const style = {
  container: {
    minHeight: height - width / 2,
  },
};

const DaySummaries = ({reports}) => (
  <ScrollView style={style.container}>
    {reports &&
      sortReports(reports).map(([date, report]) => (
        <DaySummary key={date} date={date} circles={report.circles} />
      ))}
  </ScrollView>
);

DaySummaries.propTypes = {
  reports: PropTypes.object.isRequired,
};

export default DaySummaries;
