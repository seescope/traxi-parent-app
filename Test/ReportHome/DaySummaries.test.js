import React from 'react';
import renderer from 'react-test-renderer';
import DaySummaries from '../../App/ReportHome/DaySummaries';

it('Renders a list of day summaries', () => {
  const REPORT_DATA = {
    '2016-12-10': {
      circles: [],
    },
    '2016-12-11': {
      circles: [],
    },
    '2016-12-12': {
      circles: [],
    },
    '2016-12-13': {
      circles: [],
    },
  };

  const tree = renderer.create(<DaySummaries reports={REPORT_DATA} />).toJSON();
  expect(tree).toMatchSnapshot();
});
