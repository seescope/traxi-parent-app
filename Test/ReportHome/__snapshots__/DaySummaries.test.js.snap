exports[`test Renders a list of day summaries 1`] = `
<ScrollView
  style={
    Object {
      "minHeight": 959,
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 16,
          "width": 750,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFF",
            "fontFamily": "Raleway-ExtraBold",
            "fontSize": 30,
            "marginBottom": -2,
            "textAlign": "left",
          }
        }>
        Tuesday
      </Text>
      <View
        elevation={11}
        style={
          Object {
            "height": 96,
            "paddingTop": 6,
            "width": 96,
          }
        }>
        <Image
          source={undefined}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#4CAF50",
              "borderRadius": 42,
              "height": 84,
              "justifyContent": "center",
              "left": 6,
              "width": 84,
              "zIndex": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(244, 67, 54)",
                "fontSize": 23,
                "fontWeight": "bold",
              }
            }>
            undefinedm
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(75, 73, 73, 1)",
                "fontSize": 12,
                "fontWeight": "200",
              }
            } />
        </Image>
        <View />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 16,
          "width": 750,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFF",
            "fontFamily": "Raleway-ExtraBold",
            "fontSize": 30,
            "marginBottom": -2,
            "textAlign": "left",
          }
        }>
        Monday
      </Text>
      <View
        elevation={11}
        style={
          Object {
            "height": 96,
            "paddingTop": 6,
            "width": 96,
          }
        }>
        <Image
          source={undefined}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#4CAF50",
              "borderRadius": 42,
              "height": 84,
              "justifyContent": "center",
              "left": 6,
              "width": 84,
              "zIndex": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(244, 67, 54)",
                "fontSize": 23,
                "fontWeight": "bold",
              }
            }>
            undefinedm
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(75, 73, 73, 1)",
                "fontSize": 12,
                "fontWeight": "200",
              }
            } />
        </Image>
        <View />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 16,
          "width": 750,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFF",
            "fontFamily": "Raleway-ExtraBold",
            "fontSize": 30,
            "marginBottom": -2,
            "textAlign": "left",
          }
        }>
        Sunday
      </Text>
      <View
        elevation={11}
        style={
          Object {
            "height": 96,
            "paddingTop": 6,
            "width": 96,
          }
        }>
        <Image
          source={undefined}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#4CAF50",
              "borderRadius": 42,
              "height": 84,
              "justifyContent": "center",
              "left": 6,
              "width": 84,
              "zIndex": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(244, 67, 54)",
                "fontSize": 23,
                "fontWeight": "bold",
              }
            }>
            undefinedm
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(75, 73, 73, 1)",
                "fontSize": 12,
                "fontWeight": "200",
              }
            } />
        </Image>
        <View />
      </View>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "padding": 16,
          "width": 750,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#FFF",
            "fontFamily": "Raleway-ExtraBold",
            "fontSize": 30,
            "marginBottom": -2,
            "textAlign": "left",
          }
        }>
        Saturday
      </Text>
      <View
        elevation={11}
        style={
          Object {
            "height": 96,
            "paddingTop": 6,
            "width": 96,
          }
        }>
        <Image
          source={undefined}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFF",
              "borderColor": "#4CAF50",
              "borderRadius": 42,
              "height": 84,
              "justifyContent": "center",
              "left": 6,
              "width": 84,
              "zIndex": 2,
            }
          }>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgb(244, 67, 54)",
                "fontSize": 23,
                "fontWeight": "bold",
              }
            }>
            undefinedm
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "rgba(75, 73, 73, 1)",
                "fontSize": 12,
                "fontWeight": "200",
              }
            } />
        </Image>
        <View />
      </View>
    </View>
  </View>
</ScrollView>
`;
